src/layout/constraint.rs

```diff
@@ -364,6 +366,18 @@ impl Constraint {
             .map(Constraint::Proportional)
             .collect_vec()
     }
+
+    pub fn rank(&self) -> usize {
```

Comment:
Is this really supposed to be public? If so could we document it?

Reply:
Good catch, it shouldn't be public.